### PR TITLE
testing/softhsm: new aport

### PR DESCRIPTION
**Pull request by silveraid**
http://www.softhsm.org/
cryptographic store accessible through a PKCS #11

### CODE REVIEW

File: testing/softhsm/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+build() {
+        cd "$builddir"
+        ./configure \
+                --prefix=/usr \
+                --sysconfdir=/etc \
+                --localstatedir=/var \
+                || return 1
+        make -j8 || return 1
```
Comment (ncopa) [inline testing/softhsm/APKBUILD]: we should not specify `-j8` to make. The build server should have a good number set via `MAKEFLAGS` env var. Some builder will have higher number and some will have lower.

File: testing/softhsm/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+depends="openssl"
+makedepends="openssl-dev automake autoconf libtool file"
+install=""
+subpackages="$pkgname-doc"
+source="softhsm-$pkgver.tar.gz::https://github.com/opendnssec/SoftHSMv2/archive/$pkgver.tar.gz"
+builddir="$srcdir/SoftHSMv2-2.3.0"
+
+prepare() {
+        cd "$builddir"
+        sh autogen.sh || return 1
```
Comment (ncopa) [inline testing/softhsm/APKBUILD]: we dont need the `|| return 1`

File: testing/softhsm/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+build() {
+        cd "$builddir"
+        ./configure \
+                --prefix=/usr \
+                --sysconfdir=/etc \
+                --localstatedir=/var \
+                || return 1
```
Comment (ncopa) [inline testing/softhsm/APKBUILD]: the `|| return 1` should be removed

File: testing/softhsm/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+# Maintainer: Frank Felhoffer <silveraid@hackme.ca>
+pkgname=softhsm
+pkgver=2.3.0
+pkgrel=0
+pkgdesc="cryptographic store accessible through a PKCS #11"
+url="http://www.softhsm.org/"
+arch="all"
+license="BSD2"
+depends="openssl"
+makedepends="openssl-dev automake autoconf libtool file"
```
Comment (ncopa) [inline testing/softhsm/APKBUILD]: Does this build with `libressl-dev`? If not, can you please file a bug upstream? We want avoid openssl if possible.

Comment (silveraid) [reply testing/softhsm/APKBUILD]: I was able to compile it with libressl without any problem.

File: testing/softhsm/APKBUILD (hunk 2 above)
Comment (silveraid) [reply testing/softhsm/APKBUILD]: Removed.

File: testing/softhsm/APKBUILD (hunk 3 above)
Comment (silveraid) [reply testing/softhsm/APKBUILD]: Removed.

File: testing/softhsm/APKBUILD (hunk 1 above)
Comment (silveraid) [reply testing/softhsm/APKBUILD]: Removed.

**Comment by silveraid** [comment]
All requested changes have been completed.

**Comment by gbolo** [comment]
+1 on this new aport! Considering alpine is _"security-oriented"_, it would be nice to have some **pkcs11** related packages available!

File: testing/softhsm/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        sh autogen.sh
+}
+
+build() {
+        cd "$builddir"
+        ./configure \
+                --prefix=/usr \
+                --sysconfdir=/etc \
+                --localstatedir=/var \
+                --disable-gost
```
Comment (fabled) [inline testing/softhsm/APKBUILD]: Could you add `--disable-static` too? Otherwise the large and useless `.a` gets bundled in.

Comment (silveraid) [reply testing/softhsm/APKBUILD]: Added, thanks for the suggestion.

**Comment by silveraid** [comment]
All requested changes have been completed.

**Pull request closed without merging**